test/resources/auction_examples.js

```diff
@@ -0,0 +1,366 @@
+/* eslint-disable no-console */
+
+const BN = require("bn.js")
+const assert = require("assert")
+
+// fee is (1 / feeDenominator)
+const feeDenominator = new BN(1000)
+const feeDenominatorMinusOne = feeDenominator.sub(new BN(1))
+
+// amount epsilon is used to account for rounding errors in the amount
+// calculations so that we don't underflow on limit term calculation
+const amountEpsilon = new BN(999000)
+
+function toETH(value) {
+  const oneFinney = new BN(10).pow(new BN(15))
+  return new BN(value * 1000).mul(oneFinney)
+}
+
+function feeSubtracted(x, n) {
+  assert(BN.isBN(x))
+  const res = x.mul(feeDenominatorMinusOne).div(feeDenominator)
+  if (!n || n == 1) {
+    return res
+  }
+  return feeSubtracted(res, n - 1)
+}
+
+function feeAdded(x) {
+  assert(BN.isBN(x))
+  return x.mul(feeDenominator).div(feeDenominatorMinusOne)
+}
+
+function getExecutedSellAmount(executedBuyAmount, buyTokenPrice, sellTokenPrice) {
+  // assert([executedBuyAmount, buyTokenPrice, sellTokenPrice].all(BN.isBN))
```

Comment:
Can we get this assert?

Reply:
Get it in or out?

Reply:
I will do this later... when we use the new math file.